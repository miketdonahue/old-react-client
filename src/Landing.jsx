import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { shape, string, func } from 'prop-types';
import { setSearchTerm } from './actionCreators';

// Do not pass around the entire state. Just what you need.
const mapStateToProps = state => ({ searchTerm: state.searchTerm });

// This passes events that happen in the UI which translate to actions in Redux
// Passing events to Redux
const mapDispatchToProps = dispatch => ({
  handleSearchTermChange(event) {
    // Dispatched to redux
    dispatch(setSearchTerm(event.target.value));
  },
});

class Landing extends Component {
  goToSearch = (event) => {
    event.preventDefault();
    this.props.history.push('/search'); // All routes have history injected on props
  };

  render() {
    return (
      <div className="landing">
        <h1>TV Shows</h1>
        <form onSubmit={this.goToSearch}>
          <input type="text" placeholder="Search" value={this.props.searchTerm} onChange={this.props.handleSearchTermChange} />
        </form>
        <Link to="/search">or Browse All</Link>
      </div>
    );
  }
}

Landing.defaultProps = {
  searchTerm: '',
  handleSearchTermChange: function noop() {},
  history: {},
};

Landing.propTypes = {
  searchTerm: string,
  handleSearchTermChange: func,
  history: shape(),
};

export default connect(mapStateToProps, mapDispatchToProps)(Landing);
