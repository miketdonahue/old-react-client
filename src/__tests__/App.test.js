import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Landing from '../pages/Landing';

test('<Landing />', () => {
  const component = shallow(<Landing />);

  expect(toJson(component)).toMatchSnapshot();
});
