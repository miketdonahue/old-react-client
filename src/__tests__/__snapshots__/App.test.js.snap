// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> 1`] = `
<div>
  Hello, world!!!
</div>
`;
